app/src/main/java/com/cp/tasky/auth/register/presentation/RegisterScreen.kt

```diff
@@ -0,0 +1,208 @@
+package com.cp.tasky.auth.register.presentation
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material3.FilledIconButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.cp.tasky.R
+import com.cp.tasky.core.data.util.Error
+import com.cp.tasky.core.data.util.Result
+import com.cp.tasky.ui.sharecomponent.CallToActionButton
+import com.cp.tasky.ui.sharecomponent.PasswordTextField
+import com.cp.tasky.ui.sharecomponent.UserInputTextField
+import com.cp.tasky.ui.theme.EXTRA_LARGE_32_DP
+import com.cp.tasky.ui.theme.LARGE_16_DP
+import com.cp.tasky.ui.theme.MAJOR_EXTRA_LARGE_64_DP
+import com.cp.tasky.ui.theme.MINOR_EXTRA_LARGE_48_DP
+
+@Composable
+fun RegisterScreenRoot(
+    modifier: Modifier = Modifier,
+    registerViewModel: RegisterViewModel = hiltViewModel(),
+) {
+    RegisterScreen(
+        modifier = modifier,
+        viewState = registerViewModel.registerScreenState,
+        networkState = registerViewModel.registerScreenNetworkState.collectAsStateWithLifecycle().value,
+        onEvents = registerViewModel::onEvents
+    )
+}
+
+@Composable
+fun RegisterScreen(
+    modifier: Modifier = Modifier,
```

Comment:
Optional parameters should come after the mandatory ones, so you can pass the mandatory ones without being forced to use named arguments (even though I would still use that in most cases)